### ISSUE TITLE
Rule proposal: `vue/no-unused-prop-declarations`

### ISSUE THREAD

**Issue by cesaralvrz** [vuejs/eslint-plugin-vue#2298]
**Please describe what the rule should do:**
Warn if a prop is defined, but not used inside the component.

**What category should the rule belong to?**

- [ ] Enforces code style (layout)
- [X] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<!-- ✅ GOOD -->
<template>
    <div>Hello World</div>
</template>

<script setup>
const props = defineProps(['foo'])

console.log(props.foo)
</script>


<!-- ❌ BAD -->
<template>
    <div>Hello World</div>
</template>

<script setup>
const props = defineProps(['foo'])
</script>
```

```vue
<!-- ✅ GOOD -->
<template>
    <div>{{ foo }}</div>
</template>

<script>
export default {
  props: ['foo'],
}
</script>


<!-- ❌ BAD -->
<template>
    <div>Hello World</div>
</template>

<script>
export default {
  props: ['foo'],
}
</script>
```

**Additional context**

The main objective is to be able to get rid of code that is no longer used.

**Comment by FloEdelmann** [comment]
This should already be covered by https://eslint.vuejs.org/rules/no-unused-properties.html. Isn't it?

**Comment by cesaralvrz** [comment]
Actually it is, sorry for missing it. Thank you!

**Comment by FloEdelmann** [comment]
Feel free to submit a PR to improve the documentation of that rule (or another rule that should link to it), so it's easier to find using your keywords.
